tests/rparWith_r0.hs

```diff
@@ -0,0 +1,27 @@
+-- | Test if `rparWith r0` has the documented behaviour: to create a spark,
+-- that does nothing. In particular the spark won't evaluate to weak head
+-- normal form (WHNF).
+module Main (main) where
+
+import Control.Concurrent (threadDelay)
+import Control.Exception (evaluate)
+import System.IO.Unsafe (unsafePerformIO)
+
+import Control.Parallel.Strategies
+
+data LazyBox a = LazyBox a
+
+main = do
+  -- Create spark, that must *not* evaluate printWhenEvaluated
+  LazyBox printWhenEvaluated <- evaluate $ runEval $ do
+    let printWhenEvaluated = unsafePerformIO $ putStrLn "Must be printed at the end."
+    sparkedResult <- rparWith r0 printWhenEvaluated
+    return $ LazyBox sparkedResult
```

Comment:
Since #3 has been accepted, one can simplify this mess after merge to master:
```Haskell
printWhenEvaluated <- unsafePerformIO (putStrnLn "Must be printed at the end.") `usingIO` rparWith r0
```